androidApp/src/main/java/com/example/tasky/android/alarm/di/AlarmModule.kt

```diff
@@ -2,14 +2,17 @@ package com.example.tasky.android.alarm.di
 
 import com.example.tasky.alarm.data.AlarmRepository
 import com.example.tasky.alarm.domain.IAlarmRepository
+import com.example.tasky.alarm.domain.IAlarmScheduler
 import com.example.tasky.android.alarm.data.AlarmScheduler
-import com.example.tasky.android.alarm.domain.IAlarmScheduler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
 import org.koin.core.module.dsl.singleOf
 import org.koin.dsl.bind
 import org.koin.dsl.module
 
 val alarmModule =
     module {
+        factory { CoroutineScope(Dispatchers.IO) }
```

Comment:
This scope’s lifecycle will be tied to the class where this is injected, which is not what you want 😅 
This is how you can set up the DI for a scope tied to the `Application`:

```
class MyApp: Application() {
    val applicationScope = CoroutineScope(SupervisorJob())
    
    // ...
}
```
We use `SupervisorJob()` so that if any coroutine launched from this scope throws an exception, it won’t affect the other coroutines (children) launched from the scope (parent).

You can then provide an instance of this scope in your Koin module:
```
val appModule = module {
  single<CoroutineScope> {
    // androidApplication() is an extension function on the Scope provided from single {}
    (androidApplication() as TaskyApp).applicationScope
  }
}
```
Then, inject it into the class you need it in:
```
class MyRepository(
    private val applicationScope: CoroutineScope,
    ...
)
```